libvast/vast/store.hpp

```diff
@@ -13,39 +13,33 @@
 
 #pragma once
 
-#include <vector>
+#include <caf/fwd.hpp>
 
-#include "vast/expected.hpp"
-#include "vast/ids.hpp"
+#include <caf/expected.hpp>
 
-namespace vast {
-
-class event;
+#include "vast/fwd.hpp"
 
-// TODO: Instead of using vector<event> as an abstraction for a set of events,
-// we'd like to use vast::batch here eventually. However, this requires first
-// an update of the batch interface/semantics, which is part of a separate
-// story.
+namespace vast {
 
 /// A key-value store for events.
 class store {
 public:
-  virtual ~store() = default;
+  virtual ~store();
```

Comment:
Why did you move the implementation to a separate file? Imho if it is default it should be defined at such. If the definition is out of line I expect it to be nontrivial.

Reply:
The reason is to avoid weak vtables. @Neverlord hit me so frequently on my fingers for writing `= default` to a virtual destructor, that I fear the pain each time I catch myself writing it again.

Reply:
😈